File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> basket = new Stack<>();
        for(int i = 0; i < moves.length; i++) {
        	for(int j = 0; j < board.length; j++) {
        		if(board[j][moves[i]-1] != 0) {
        			basket.push(board[j][moves[i]-1]);
        			board[j][moves[i]-1] = 0; 
        			break;
        		}
        	}
        	if(basket.size() > 1 && basket.get(basket.size()-1) == basket.get(basket.size()-2)) {
        		basket.pop();basket.pop();
        		answer+=2;
        	}
        }
        return answer;
    }
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let answer = 0;
    let stack = [];
    moves.forEach(move => {
        let pick;
        board.some(row => {
            if (row[move - 1]) {
                pick = row[move - 1];
                row[move - 1] = 0;
                return true;
            } 
        })
        if (!pick) return;
        if (stack[stack.length - 1] === pick) {
            stack.pop();
            answer += 2;
        } else stack.push(pick);
    })
    return answer;
}